src/services/supabase.ts

```diff
@@ -195,25 +210,22 @@ export const endSubscription = (subscription: RealtimeSubscription) => {
         .catch(() => {});
 };
 
-export const endAllSubscriptions = () => {
-    return supabaseClient
-        .removeAllSubscriptions()
-        .then(() => {})
-        .catch(() => {});
-};
-
 export const startSubscription = (
-    query: any,
+    query: SupabaseQueryBuilder<any>,
     success: Function,
     failure: Function,
     keepSubscription?: boolean
 ) => {
     const subscription = query
-        .on('*', async (payload: any) => {
+        .on('*', async (payload: SupabaseRealtimePayload<any>) => {
             const response = payload.new
                 ? payload.new
-                : payload.record
-                ? payload.record
+                : // TODO (typing) during manual testing I have seen record be in the response
+                //      even though the type says it is not there. Needs more research
+                // eslint-disable-next-line @typescript-eslint/dot-notation
+                payload['record']
+                ? // eslint-disable-next-line @typescript-eslint/dot-notation
+                  payload['record']
```

Comment:
Wanna look into this but it feels like the Supabase types aren't totally accurate.